Dockerfile

```diff
@@ -1,11 +1,12 @@
-FROM golang:1.22
+FROM --platform=$BUILDPLATFORM golang:1.22
 WORKDIR /go/src
 # make deps fetching cacheable
 COPY go.mod go.sum ./
 RUN go mod download
 # build
 COPY . .
-RUN make build
+ARG TARGETARCH
+RUN GOARCH=$TARGETARCH make build
 
 # build real cloud-provider-kind image
 FROM docker:25.0-dind
```

Comment:
will make a seperate PR, but we have no reason to actually ship the daemon right? this image is bigger than we need, we just need the `docker` CLI, right? or am I missing something.

Reply:
we do, the cloud provider kind needs docker when running inside docker

Reply:
confirmed we only need the CLI with @aojea

Reply:
filed as another PR so we can decide separately https://github.com/kubernetes-sigs/cloud-provider-kind/pull/174, but included this existing commit to avoid diff since it's about to merge hopefully